### ISSUE TITLE
Transmittance depth guided loss

### ISSUE THREAD

**Issue by vgrau98** [nerfstudio-project/nerfstudio#2324]
Hi, 

I am implementing text2nerf from the following paper: https://arxiv.org/pdf/2305.11588.pdf, and I have a few questions about the cost function used that might be interesting for depth-driven training. 

 The loss is defined as: 
<img width="162" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/2964f7e1-7874-452d-ba8c-179b75577c14">

With:
<img width="105" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/f82d7a8c-06db-4079-a706-9a41955a89b0">

where m(t) is a mask indicator that satisfies m(t) = 1 when t < z, otherwise m(t) = 0. z is the pixel-wise depth value

1. First, I don't understand how L_T can be injected directly into the global loss, given that we want to maximise the norm 
<img width="63" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/fdbdb828-c392-411c-950a-fd3aa3ce8593">

We want the transmittance to be max before the depth, and min after. 

2. I modified the transmittance loss, being MSELoss(target,  $`l_{2}(m(t)*T(t))`$) with target = 1 when t<depth, and 0 otherwise. But my results are strange. I have a tensorf field with the following loss weights (same as the paper): 

rgb_loss: 1, depth_loss: 0.005 and depth_transmittance_loss: 1000

Rgb render from nerfstudio viewer:

<img width="652" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/09dd2e87-7527-4d02-9840-85d8ff43f647">

<img width="644" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/4466afc9-8535-4d23-9910-3d4d5e664bff">

Without the transmittance loss results are more consistent but there are still some artifacts (sorry i did not use the same rgb input frame):

<img width="416" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/457ba4d1-39cd-44f1-aa8c-6db4d553a78d">

<img width="645" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/35843843/13bf1cbd-675b-41a3-9a41-fd1e6e104a12">

The depth map used comes from the script : https://github.com/thygate/stable-diffusion-webui-depthmap-script
It is in meters and normalized according to the nerfstudio scaling factor (scene in unit cube). 

Here are the configs used:

TrainerConfig:
    _target: <class 'nerfstudio.engine.trainer.Trainer'>
    output_dir: outputs
    method_name: tensorf-text2nerf
    experiment_name: None
    project_name: nerfstudio-project
    timestamp: {timestamp}
    machine: MachineConfig:
        seed: 42
        num_devices: 1
        num_machines: 1
        machine_rank: 0
        dist_url: auto
        device_type: cuda
    logging: LoggingConfig:
        relative_log_dir: .
        steps_per_log: 10
        max_buffer_size: 20
        local_writer: LocalWriterConfig:
            _target: <class 'nerfstudio.utils.writer.LocalWriter'>
            enable: True
            stats_to_track: [EventName.ITER_TRAIN_TIME
            EventName.TRAIN_RAYS_PER_SEC
            EventName.CURR_TEST_PSNR
            EventName.VIS_RAYS_PER_SEC
            EventName.TEST_RAYS_PER_SEC
            EventName.ETA]
            max_log_size: 10
        profiler: basic
    viewer: ViewerConfig:
        relative_log_filename: viewer_log_filename.txt
        websocket_port: 7007
        websocket_port_default: 7007
        websocket_host: 0.0.0.0
        num_rays_per_chunk: 32768
        max_num_display_images: 512
        quit_on_train_completion: True
        image_format: jpeg
        jpeg_quality: 90
    pipeline: VanillaPipelineConfig:
        _target: <class 'nerfstudio.pipelines.base_pipeline.VanillaPipeline'>
        datamanager: StreamDataManagerConfig:
            _target: <class 'nerfstudio.data.datamanagers.stream_datamanager.StreamDataManager'>
            data: None
            camera_optimizer: CameraOptimizerConfig:
                _target: <class 'nerfstudio.cameras.camera_optimizers.CameraOptimizer'>
                mode: off
                position_noise_std: 0.0
                orientation_noise_std: 0.0
                optimizer: AdamOptimizerConfig:
                    _target: <class 'torch.optim.adam.Adam'>
                    lr: 0.0006
                    eps: 1e-15
                    max_norm: None
                    weight_decay: 0
                scheduler: ExponentialDecaySchedulerConfig:
                    _target: <class 'nerfstudio.engine.schedulers.ExponentialDecayScheduler'>
                    lr_pre_warmup: 1e-08
                    lr_final: None
                    warmup_steps: 0
                    max_steps: 10000
                    ramp: cosine
                param_group: camera_opt
            masks_on_gpu: None
            train_num_rays_per_batch: 4096
            train_num_images_to_sample_from: -1
            train_num_times_to_repeat_images: -1
            eval_num_rays_per_batch: 4096
            eval_num_images_to_sample_from: -1
            eval_num_times_to_repeat_images: -1
            eval_image_indices: [0]
            collate_fn: <function nerfstudio_collate at 0x7fb29c66e3b0>
            camera_res_scale_factor: 1.0
            patch_size: 1
            total_nbr_views: 226
        model: TensoRFModelT2NConfig:
            _target: <class 'nerfstudio.models.tensorf_text2nerf_model.TensoRFModelT2N'>
            enable_collider: True
            collider_params: {'near_plane': 0.5, 'far_plane': 6}
            loss_coefficients: {'rgb_loss': 1.0, 'tv_reg_density': 0, 'tv_reg_color': 0, 'l1_reg': 0, 'depth_loss': 0.005, 'depth_transmittance': 0.0}
            eval_num_rays_per_chunk: 32768
            prompt: None
            init_resolution: 128
            final_resolution: 300
            upsampling_iters: [2000
            3000
            4000
            5500
            7000]
            num_samples: 50
            num_uniform_samples: 100
            num_den_components: 16
            num_color_components: 48
            appearance_dim: 27
            tensorf_encoding: vm
            regularization: l1
    optimizers: {'fields': {'optimizer': AdamOptimizerConfig(_target=<class 'torch.optim.adam.Adam'>, lr=0.001, eps=1e-08, max_norm=None, weight_decay=0), 'scheduler': ExponentialDecaySchedulerConfig(_target=<class 'nerfstudio.engine.schedulers.ExponentialDecayScheduler'>, lr_pre_warmup=1e-08, lr_final=0.0001, warmup_steps=0, max_steps=30000, ramp='cosine')}, 'encodings': {'optimizer': AdamOptimizerConfig(_target=<class 'torch.optim.adam.Adam'>, lr=0.02, eps=1e-08, max_norm=None, weight_decay=0), 'scheduler': ExponentialDecaySchedulerConfig(_target=<class 'nerfstudio.engine.schedulers.ExponentialDecayScheduler'>, lr_pre_warmup=1e-08, lr_final=0.002, warmup_steps=0, max_steps=30000, ramp='cosine')}}
    vis: viewer+tensorboard
    data: None
    prompt: None
    relative_model_dir: nerfstudio_models
    load_scheduler: True
    steps_per_save: 2000
    steps_per_eval_batch: 500
    steps_per_eval_image: 500
    steps_per_eval_all_images: 25000
    max_num_iterations: 1000
    mixed_precision: True
    use_grad_scaler: False
    save_only_latest_checkpoint: True
    load_dir: None
    load_step: None
    load_config: None
    load_checkpoint: None
    log_gradients: False
    gradient_accumulation_steps: 1

So my question is about the cost function with transmittance, and why do I get this kind of result? Any hints would be greatly appreciated, thank you!
